homework/index.js

```diff
@@ -29,23 +31,160 @@
     return elem;
   }
 
+  function convertDateTime(dateTimeTxt) {
+    const dateTime = new Date(dateTimeTxt);
+    return dateTime.toLocaleString();
+  }
+
+  function sortRepositoriesByName(a, b) {
+    return a.name.localeCompare(b.name);
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    ul.textContent = ' ';
+    const repoList = createAndAppend('li', ul, { class: 'rep-list' });
+    const repositoryName = createAndAppend('p', repoList, {
+      text: ' Repository: ',
+      class: 'repo-details',
+    });
+
+    const description = createAndAppend('p', repoList, {
+      text: ' Description: ',
+      class: 'repo-details',
+    });
+
+    const forkedNumbers = createAndAppend('p', repoList, {
+      text: ' Forks: ',
+      class: 'repo-details',
+    });
+
+    const updatedTime = createAndAppend('p', repoList, {
+      text: ' Updated: ',
+      class: 'repo-details',
+    });
+
+    createAndAppend('a', repositoryName, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+
+    if (repo.description === null) {
+      createAndAppend('span', description, {
+        text: 'No Description Added Yet',
+        class: 'texts',
+      });
+    } else {
+      createAndAppend('span', description, {
+        text: repo.description,
+        class: 'texts',
+      });
+    }
+
+    createAndAppend('span', forkedNumbers, {
+      text: repo.forks,
+      class: 'texts',
+    });
+
+    createAndAppend('span', updatedTime, {
+      text: convertDateTime(repo.updated_at),
+      class: 'texts',
+    });
+  }
+
+  function getContributorsInfo(repositories, ul) {
+    ul.textContent = ' ';
+    const url = repositories.contributors_url;
+    fetchJSON(url)
+      .then(contributions => {
+        contributions.forEach(contributor => {
```

Comment:
Shorter function are preferred over longer functions, as it improves the readability of your code. In this case, I would replace the arrow function for this `.forEach()` with a named function like this:

```js
function renderContributor(contributor, ul) {
  const li = createAndAppend('li', ul, { class: 'contributors-li' });
  const contributorImage = createAndAppend('div', li, {
    class: 'contributor-image',
  });
  ...
}

function getContributorsInfo(repositories, ul) {
  ul.textContent = '';
  const url = repositories.contributors_url;
  fetchJSON(url)
    .then(contributions => {
      contributions.forEach(contributor =>
        renderContributor(contributor, ul),
      );
    })
   ...
}
```

Reply:
I changed the code based on the feedback, hopefully it is correct :), thank you @remarcmij